### PR TITLE
RE-1316 Implement Sandbox Integration Gating Job

### PR DESCRIPTION
**Pull request by dealako**
- Created l2 job template.

Issue: [RE-1425](https://rpc-openstack.atlassian.net/browse/RE-1425)

### CODE REVIEW

File: rpc_jobs/standard_job_l2_gate.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+- project:
+    name: "l2_gate_tests"
```
Comment (mattt416) [inline rpc_jobs/standard_job_l2_gate.yml]: I'd also flip this name to something like `rpc-product-1-gate` as the project names will have to be unique across all components.

File: rpc_jobs/standard_job_l2_gate.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          SLAVE_CONTAINER_DOCKERFILE_BUILD_ARGS: "{SLAVE_CONTAINER_DOCKERFILE_BUILD_ARGS}"
+      - string:
+          name: skip_pattern
+          default: "{skip_pattern}"
+          description: |
+            Python re compatible regex, with verbose flag enabled. If all files changed by the
+            pull request match the regex the build will exit without running the test scripts.
+            This is used to skip tests that are not relevant to a change, for example testing a
+            deployment when only changing documentation. By default no builds are skipped.
+    triggers:
```
[the remark this answers is not in the document]
Comment (dealako) [reply rpc_jobs/standard_job_l2_gate.yml]: Removed...

File: rpc_jobs/standard_job_l2_gate.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    scenario:
+      - "lint"
+    action:
+      - "test"
+    jobs:
+      - 'L2_{repo_name}-{series}-{image}-{scenario}-{action}'
+    jira_project_key: 'RE'
+
+- job-template:
+    name: 'L2_{repo_name}-{series}-{image}-{scenario}-{action}'
```
[the remark this answers is not in the document]
Comment (dealako) [reply rpc_jobs/standard_job_l2_gate.yml]: Gimme a good project name:

File: rpc_jobs/standard_job_l2_gate.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          trigger-phrase: '.*recheck_(cit_)?all.*|.*recheck_(cit_)?{image}_{scenario}_{action}.*'
+          only-trigger-phrase: "{obj:TRIGGER_PR_PHRASE_ONLY}"
+          white-list-target-branches: "{branches}"
+          auth-id: "github_account_rpc_jenkins_svc"
+          status-context: 'CIT/{image}_{scenario}_{action}'
+          cancel-builds-on-update: true
+
+    dsl: |
+      library "rpc-gating@${{RPC_GATING_BRANCH}}"
+      common.stdJob("l2_gate_tests", "{credentials}", "{jira_project_key}")
```
[the remark this answers is not in the document]
Comment (dealako) [reply rpc_jobs/standard_job_l2_gate.yml]: This assumes only one "gate"?

File: rpc_jobs/standard_job_l2_gate.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dealako) [reply rpc_jobs/standard_job_l2_gate.yml]: I see your comment below, "rpc-product-1-gate" it is...

File: rpc_jobs/standard_job_l2_gate.yml (hunk 1 above)
Comment (dealako) [reply rpc_jobs/standard_job_l2_gate.yml]: Done

File: rpc_jobs/standard_job_gate.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    jira_project_key: 'RE'
+
+- job-template:
+    name: 'GATE_{repo_name}-{series}-{image}-{scenario}-{action}'
+    series: "master"
+    project-type: pipeline
+    concurrent: true
+    FLAVOR: "performance1-1"
+    IMAGE: "Ubuntu 16.04 LTS (Xenial Xerus) (PVHVM)"
+    TRIGGER_PR_PHRASE_ONLY: false
```
[the remark this answers is not in the document]
Comment (dealako) [reply rpc_jobs/standard_job_gate.yml]: Removed!

**Pull request merged**